### PR TITLE
[FIX #2530] Don't expect `.parseJail` to return contents of `_status_catalog`.

### LINKED ISSUE #2530: Create `catalog` cell variable on status-react side
### User Story

As a developer, I'd like to move creating `catalog` variable from status-go to status-react, as status-go has no knowledge about it and why it is required.

### Description

*Type*: Feature

*Summary*: In status-go, during cell initialization, the following code is executed:
```js
var catalog = JSON.stringify(_status_catalog);
```

status-go does not know about `_status_catalog`, thus, it should not rely on it.

#### Expected behavior
There is no change in cell initialization after executing that JS fragment on status-react side.

#### Actual behavior
It is correct.

### Solution
`catalog` variable should be created on status-react side using [`ExecuteJS` binding](https://github.com/status-im/status-go/blob/develop/lib/library.go#L343-L346).

Please, comment on https://github.com/status-im/status-go/issues/450 after resolving.

### Additional Information
* Status version: All
* Operating System: Android and iOS

### PR DESCRIPTION
**Pull request by mandrigin**
### Summary:

Addresses https://github.com/status-im/status-react/issues/2530

This is my attempt to prepare the client codebase to the go library refactoring mentioned in [status-go#450](https://github.com/status-im/status-go/issues/450).

### Review notes (optional):
This is my first time trying coding in Closure, your feedback is much appreciated!
I tried to keep the compatibility and make the minimum amount of changes.
~I made it a `WIP` mainly because of feedback, I didn't really want to add anything to this PR.~

1. Add a method to get catalog to `status.js`.
2. Keep backward compatibility of `parse-jail` function.
3. Export `ExecuteJS` method to Android and iOS.

### Steps to test:
- Open Status
- Sign-in
- Expect everything to work properly.

status: done

### CODE REVIEW

**Comment by mandrigin** [comment]
Updated the title of the PR since it isn't really a WIP. Waiting for feedback.

File: modules/react-native-status/ios/RCTStatus/RCTStatus.m [hunk 1]
```diff
@@ -102,6 +102,18 @@ -(void)setBridge:(RCTBridge *)newBridge
     });
 }
 
+//////////////////////////////////////////////////////////////////// executeJS
+RCT_EXPORT_METHOD(executeJS:(NSString *)chatId
```
Comment (oskarth) [inline modules/react-native-status/ios/RCTStatus/RCTStatus.m:106]: Did you try this with existing version of `status-go` develop? I would expect this to fail if executeJS isn't exposed in status-go yet

**Comment by oskarth** [comment]
Nice one @mandrigin and good initiative!

I triggered a custom build of this in Jenkins just to see: https://jenkins.status.im/job/status-react/job/parametrized%20build/72/ (internal link only)

File: modules/react-native-status/ios/RCTStatus/RCTStatus.m (hunk 1 above)
Comment (oskarth) [reply modules/react-native-status/ios/RCTStatus/RCTStatus.m:106]: Oh so this is in status-go develop already https://github.com/status-im/status-go/blob/develop/lib/library.go#L343 :)

File: src/status_im/native_module/impl/module.cljs [hunk 2]
```diff
@@ -120,9 +120,22 @@
   (when status
     (call-module #(.discardTransaction status id))))
 
+(defn execute-js [bot-id js-code callback]
```
Comment (oskarth) [inline src/status_im/native_module/impl/module.cljs:123]: There are a few more places to add this for new bindings, you can see an example here (check src file changes): https://github.com/status-im/status-react/commit/6d5251ad2072c949c0608acde4144d4f4cff0d11

Comment (mandrigin) [reply src/status_im/native_module/impl/module.cljs:123]: thanks a lot! I'll take a look and update my PR!

**Comment by mandrigin** [comment]
@oskarth there is actually a small change proposed, that makes client-side changes much simpler: https://github.com/status-im/status-go/pull/489#issuecomment-349666570
(in a nutshell, the `parseCell` code will return the result of the code that is passed to it, so we can just pass `getCatalogJSON()` as a parameter there and then `parseCell` will return its result.)

If we do it this way, there is no need to expose `executeJS` to Closure.

I still think that `executeJS` is a nice tool anyway.
Do you want me to keep it in this PR or, maybe split it into two PRs?

**Comment by oskarth** [comment]
@mandrigin I'm not sure, but I think consensus is that we don't need `executeJS` exposed in production even if it is useful for debugging, so I'd skip this.

@rasom might be able to elaborate

**Comment by mandrigin** [comment]
@oskarth do you mean, that I should revert this fix then? It adds a binding to `executeJS`.

File: src/status_im/native_module/impl/module.cljs (hunk 2 above)
Comment (mandrigin) [reply src/status_im/native_module/impl/module.cljs:123]: added this binding to the places as in the mentioned commit

**Comment by rasom** [comment]
We discussed with @adambabik that `CreateAndInitCell` will return the value which is returned by `vm.Run`, so that no need to call `ExecuteJS`. As we don't need `ExecuteJS` for any other functionality atm better just to close PR.

**Comment by mandrigin** [comment]
@rasom ok, then I will create another PR that actually adds necessary js commands for `CreateAndInitCell` *after* this [status-go PR](https://github.com/status-im/status-go/pull/489) is merged.
In this PR I will:
1) Add JS commands to create global `catalog` variable and return it.
2) Update the references of status-go in this project.

Does it sound good?

**Pull request closed without merging**